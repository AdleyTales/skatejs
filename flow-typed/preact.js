// @flow

declare module 'preact' {
  declare export function h (a: Function | string, b: Object, ...c: Array<Object>): Object
  declare export function render (a: () => Object, b: Node, c?: Object): Object
}
