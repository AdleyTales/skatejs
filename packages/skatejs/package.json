{
  "name": "skatejs",
  "description": "Skate is a library built on top of the W3C web component specs that enables you to write functional and performant web components with a very small footprint.",
  "author": "Trey Shugart <treshugart@gmail.com> (http://treshugart.github.io)",
  "license": "MIT",
  "keywords": [
    "components",
    "custom",
    "dom",
    "elements",
    "web",
    "vdom",
    "virtual"
  ],
  "repository": "https://github.com/skatejs/skatejs",
  "version": "5.1.1",
  "files": [
    "dist/cjs",
    "dist/esm"
  ],
  "browser": "dist/esm/index.js",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "devDependencies": {
    "@skatejs/bore": "4.0.3"
  }
}
