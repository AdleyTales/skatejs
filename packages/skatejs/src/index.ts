export * from './component';
export * from './define';
export * from './name';
export * from './props';
export * from './types';
