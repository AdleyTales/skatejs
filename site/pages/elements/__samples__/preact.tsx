/** @jsx h */

import Element from '@skatejs/element';
import renderer, { h } from '@skatejs/element-preact';

class ElementPreact extends Element {
  static props = {
    name: String
  };
  name = 'World';
  render() {
    return <span>Hello, {this.name}!</span>;
  }
}

customElements.define('element-preact', ElementPreact);
