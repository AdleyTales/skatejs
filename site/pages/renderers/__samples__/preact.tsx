/** @jsx h */

import Component from 'skatejs';
import renderer, { h } from '@skatejs/renderer-preact';

class RendererPreact extends Component {
  static props = {
    name: String
  };
  renderer = renderer;
  name: string = 'World';
  render() {
    return <span>Hello, {this.name}!</span>;
  }
}

customElements.define('renderer-preact', RendererPreact);
