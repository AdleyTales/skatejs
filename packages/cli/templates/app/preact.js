const outdent = require('outdent');

module.exports = {
  'package.json': {
    dependencies: {
      '@skatejs/renderer-preact': '*',
      preact: '*'
    }
  },
  'src/index.js': outdent`
    /* @jsx h */

    import { h } from 'preact';
    import withRenderer from '@skatejs/renderer-preact';
    import { define, props, withComponent } from 'skatejs';

    @define
    class Hello extends withComponent(withRenderer()) {
      static is = 'x-hello';
      static props = {
        name: props.string
      };
      render({ name }) {
        return (
          <div>
            Hey, <strong>{name}</strong>!
          </div>
        );
      }
    }

    export default Hello;
  `
};
