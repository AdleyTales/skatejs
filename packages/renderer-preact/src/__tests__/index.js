/** @jsx h */

import { Component, h } from 'preact';
import { define } from 'skatejs';
import withRenderer from '..';

const WebComponent = withRenderer(HTMLElement);

@define
class MyElement extends WebComponent {
  render({ name }) {
    return <div>Hello, {name}!</div>;
  }
}

test('renders', () => {
  function testContent(text) {
    return `<div>Hello, ${text}!</div>`;
  }

  const el = new MyElement();
  expect(el.innerHTML).toEqual('');
  el.renderer(el, el.render.bind(el, { name: 'World' }));
  expect(el.innerHTML).toEqual(testContent('World'));
  el.renderer(el, el.render.bind(el, { name: 'Bob' }));
  expect(el.innerHTML).toEqual(testContent('Bob'));
});

test('wrappers', () => {
  class PreactComponent extends Component {
    render() {
      return <div>Hello, {this.props.children}!</div>;
    }
  }

  @define
  class PreactComponentWrapper extends WebComponent {
    constructor() {
      super();
      this.attachShadow({ mode: 'open' });
    }
    render() {
      return <PreactComponent {...this.props} />;
    }
  }

  const el = new PreactComponentWrapper();
  const { shadowRoot } = el;
  el.renderer(shadowRoot, el.render.bind(el));
  expect(shadowRoot.innerHTML).toEqual('<div>Hello, <slot></slot>!</div>');
});
