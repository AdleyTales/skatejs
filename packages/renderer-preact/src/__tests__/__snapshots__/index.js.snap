// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `""`;

exports[`renders 2`] = `"<div>Hello, World!</div>"`;

exports[`renders 3`] = `"<div>Hello, Bob!</div>"`;

exports[`wrapper - override render() 1`] = `"<div>Hello, <slot></slot>!</div>"`;

exports[`wrappers cleanup 1`] = `"<div>Hello, <slot></slot>!</div>"`;

exports[`wrappers cleanup 2`] = `""`;
